src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 5;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
+    public void resizeIfNeeded() {
+        if (elements.length == size) {
+            Object[] newArray = new Object[(int) (elements.length * 1.5)];
+            System.arraycopy(elements,0, newArray,0,size);
+            elements = newArray;
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        resizeIfNeeded();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        }
```

Comment:
lets move this check to a separate private method, because we use it several times